test/e2e/pod_infra_container_test.go

```diff
@@ -435,4 +435,15 @@ var _ = Describe("Podman pod create", func() {
 		Expect(session).Should(Exit(0))
 		Expect(session.OutputToString()).To(ContainSubstring(hostname))
 	})
+
+	It("podman pod create --share=none should not create an infra ctr", func() {
+		session := podmanTest.Podman([]string{"pod", "create", "--share", ""})
```

Comment:
I cannot find a single test for `--share none`, please add a test for this as well. You could add a for loop around this test for both values.